tests/tooling/bootstrap_ps1_test.ps1

```
@@ -0,0 +1,132 @@
+          foreach ($name in @('VCPKG_ROOT','VCPKG_DOWNLOADS','VCPKG_DISABLE_METRICS','GIT_TERMINAL_PROMPT','GCM_INTERACTIVE')) { $script:InProcessEnvironment[$name] = [Environment]::GetEnvironmentVariable($name) }
+          $script:InProcessLocation = (Get-Location).Path
+        }
+        return $status
+      }
+      $script:WrapperOutput=@(& $PowerShellExe -NoProfile -ExecutionPolicy Bypass -File (Join-Path $Repo 'bootstrap.ps1') 2>&1|Out-String); return [int]$LASTEXITCODE
+    } finally { foreach($tool in $shadowOriginal.Keys){if($null-eq$shadowOriginal[$tool]){Remove-Item -Path "Function:global:$tool" -ErrorAction SilentlyContinue}else{Set-Item -Path "Function:global:$tool" -Value $shadowOriginal[$tool]}}; foreach($n in $save.Keys){if($null -eq $save[$n]){Remove-Item "Env:$n" -ErrorAction SilentlyContinue}else{Set-Item "Env:$n" $save[$n]}} }
+  }
+  function Expect-Reject([string]$Name,[string]$Repo,[string]$Diagnostic,[hashtable]$Env=@{}) { $status=Invoke-Wrapper $Repo $Env; if($status -eq 0){Fail "$Name unexpectedly passed"}; if($script:WrapperOutput -notmatch $Diagnostic){Fail "$Name did not report its case-specific diagnostic ($Diagnostic): $script:WrapperOutput"} }
+  $canonical=Join-Path $TestRoot 'bootstrap.ps1'; if(-not(Test-Path $canonical)){Fail "bootstrap_missing_canonical_wrapper: canonical wrapper absent"}; Pass bootstrap_missing_canonical_wrapper
+  if($IsWindows -ne $true){Write-Output 'SKIP: PowerShell contract matrix requires native Windows; canonical wrapper preflight passed.'; return}
+  function bootstrap_missing_host_tool {$r=Copy-Case missing-host-tool;Write-Pin $r;Write-Fake-Tools $r;Remove-Item -LiteralPath (Join-Path $r 'fake-native-bin/cmake.cmd'),(Join-Path $r 'fake-native-bin/ctest.cmd') -Force;$isolated=Join-Path $r 'fake-native-bin';$s=Invoke-Wrapper $r @{} $isolated;if($s-eq0 -or $script:WrapperOutput-notmatch'required host tool is missing: cmake'){Fail "missing host tool failed to reject at host validation: $script:WrapperOutput"};$l=Get-Content (Join-Path $r 'fake-native.log') -Raw;if($l-match'git .*clone|bootstrap-vcpkg|vcpkg install|cmake --preset|cmake --build|ctest --preset'){Fail 'missing host tool began a later stage'};Pass $MyInvocation.MyCommand.Name}
```

PSScriptAnalyzer (1.25.0)

[info] Cmdlet 'Invoke-Wrapper' has positional parameter. Please use named parameters instead of positional parameters when calling a command.

(PSAvoidUsingPositionalParameters)